share/p2p/peers/manager.go

```diff
@@ -93,10 +93,17 @@ func NewManager(
 
 	discovery.WithOnPeersUpdate(
 		func(peerID peer.ID, isAdded bool) {
-			if isAdded && !s.peerIsBlacklisted(peerID) {
+			if isAdded {
+				if s.peerIsBlacklisted(peerID) {
+					log.Debugw("got blacklisted from discovery", "peer_id", peerID)
+					return
+				}
```

Comment:
would this ever actually happen now that we block peers?

Reply:
yes, if you block a peer externally for example

Reply:
block a peer externally via what method? We still wouldn't "discover" that peer due to conngater right?